apis/clusters/v1beta1/postgresql_types.go

```diff
@@ -75,14 +75,25 @@ type PgRestoreFrom struct {
 type PgSpec struct {
 	PgRestoreFrom         *PgRestoreFrom `json:"pgRestoreFrom,omitempty"`
 	Cluster               `json:",inline"`
-	DataCentres           []*PgDataCentre   `json:"dataCentres,omitempty"`
-	ClusterConfigurations map[string]string `json:"clusterConfigurations,omitempty"`
-	SynchronousModeStrict bool              `json:"synchronousModeStrict,omitempty"`
-	UserRefs              []*UserReference  `json:"userRefs,omitempty"`
+	DataCentres           []*PgDataCentre     `json:"dataCentres,omitempty"`
+	ClusterConfigurations map[string]string   `json:"clusterConfigurations,omitempty"`
+	SynchronousModeStrict bool                `json:"synchronousModeStrict,omitempty"`
+	UserRefs              []*NamespacedName   `json:"userRefs,omitempty"`
+	ClusterResources      ClusterResourceRefs `json:"clusterResources,omitempty"`
 	//+kubebuilder:validate:MaxItems:=1
 	ResizeSettings []*ResizeSettings `json:"resizeSettings,omitempty"`
 }
 
+type ClusterResourceRefs struct {
+	ClusterBackups                []*NamespacedName `json:"clusterBackups,omitempty"`
+	ClusterNetworkFirewallRules   []*NamespacedName `json:"clusterNetworkFirewallRules,omitempty"`
+	AWSVPCPeerings                []*NamespacedName `json:"awsVPCPeerings,omitempty"`
+	AWSSecurityGroupFirewallRules []*NamespacedName `json:"awsSecurityGroupFirewallRules,omitempty"`
+	ExclusionWindows              []*NamespacedName `json:"exclusionWindows,omitempty"`
+	GCPVPCPeerings                []*NamespacedName `json:"gcpVPCPeerings,omitempty"`
+	AzureVNetPeerings             []*NamespacedName `json:"azureVNetPeerings,omitempty"`
+}
```

Comment:
ResorceRefs must be an object for maintenance improvement

Reply:
I just think this is not an improvement but a deterioration because there are a lot of duplicate unnecessary code

Reply:
If we can simplify the code and make the resourceRefs architecture cleaner than it is using the suggested approach by Matvii, I think it will be a way better solution for now

Reply:
Ok